qiskit/providers/backend.py

```diff
@@ -229,3 +241,445 @@ class can handle either situation.
+            ValueError: if T2 is not defined for the specified qubit
+        """
+        props = self.qubit_properties(qubits)
+        if isinstance(props, QubitProperties):
+            if props.t2 is None:
+                raise ValueError(f"T2 is not defined for qubit {qubits}")
+            return props.t1
+        out_list = []
+        for index, prop in enumerate(props):
+            if prop.t2 is None:
+                raise ValueError(f"T2 is not defined for qubit {qubits[index]}")
+            out_list.append(prop.t2)
+        return np.array(out_list)
+
+    @property
+    def dt(self) -> Union[float, None]:
+        """Return the system time resolution of input signals
+
+        This is required to be implemented if the backend supports Pulse
+        scheduling.
+
+        Returns:
+            dt: The input signal timestep in seconds. If the backend doesn't
+            define ``dt`` ``None`` will be returned
+        """
+        return self.target.dt
+
+    @property
+    def dtm(self) -> float:
+        """Return the system time resolution of output signals
+
+        Returns:
+            dtm: The output signal timestep in seconds.
+
+        Raises:
+            NotImplementedError: if the backend doesn't support querying the
+                output signal timestep
+        """
+        raise NotImplementedError
+
+    @property
+    def meas_map(self) -> List[List[int]]:
+        """Return the grouping of measurements which are multiplexed
+
+        This is required to be implemented if the backend supports Pulse
+        scheduling.
+
+        Returns:
+            meas_map: The grouping of measurements which are multiplexed
+
+        Raises:
+            NotImplementedError: if the backend doesn't support querying the
+                measurement mapping
+        """
+        raise NotImplementedError
+
+    @property
+    def instruction_schedule_map(self):
+        """Return the :class:`~qiskit.pulse.InstructionScheduleMap` for the
+        instructions defined in this backend's target."""
+        return self.target.instruction_schedule_map()
+
+    def qubit_properties(
+        self, qubit: Union[int, List[int]]
+    ) -> Union[QubitProperties, List[QubitProperties]]:
+        """Return QubitProperties for a given qubit.
+
+        If there are no defined or the backend doesn't support querying these
+        details this method does not need to be implemented.
+
+        Args:
+            qubit: The qubit to get the
+                :class:`~qiskit.provider.QubitProperties` object for. This can
+                be a single integer for 1 qubit or a list of qubits and a list
+                of :class:`~qiskit.provider.QubitProperties` objects will be
+                returned in the same order
+
+        raises:
+            NotImplementedError: if the backend doesn't support querying the
+                qubit properties
+        """
+        raise NotImplementedError
+
+    def drive_channel(self, qubit: int):
```

Comment:
It seems a lot of the pulse specific functionality here may not be needed for all backends (specifically third party providers that don't provide pulse access, and circuit simulators).

Would it make more sense to have a more minimal Backend class, and then a separate abstract PulseBackend (or pulse backend mixin) that adds the additional methods needed for pulse supporting backends?

Reply:
I limited the pulse specific things to just be what was required or optional for the pulse scheduler or the pulse builder to work with a backend. In the case of `drive_channel` it's required for the pulse builder: https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/pulse/builder.py#L1358 But none of these are required for an implementation of a backend they're all optional (ie not abstract methods). I'd like to avoid the split between a pulse backend and a circuit backend as where we've done that split in the past it ends up getting pretty difficult to figure out the balance of what goes where. I think just making the things which are required for pulse use cases optional and documenting how they're used makes the most sense